### PR TITLE
SPM

### PR DESCRIPTION
**Pull request by rex-remind101**
Support Swift Package Manager.

Tests running beautifully from the command line, still need to have a CI path for these though but will do that separately.
```
$ swift test
Compile Swift Module 'AutoGraphTests' (11 sources)
Linking ./.build/x86_64-apple-macosx10.10/debug/AutoGraphPackageTests.xctest/Contents/MacOS/AutoGraphPackageTests
Test Suite 'All tests' started at 2018-07-18 14:04:59.071
Test Suite 'AutoGraphPackageTests.xctest' started at 2018-07-18 14:04:59.071
Test Suite 'AlamofireClientTests' started at 2018-07-18 14:04:59.071
Test Case '-[AutoGraphTests.AlamofireClientTests testAuthenticatingSetsTokens]' started.
Test Case '-[AutoGraphTests.AlamofireClientTests testAuthenticatingSetsTokens]' passed (0.142 seconds).
Test Case '-[AutoGraphTests.AlamofireClientTests testAuthHandlerDelegateCallsBackWhenAuthenticatingDirectlyIfRefreshing]' started.
Test Case '-[AutoGraphTests.AlamofireClientTests testAuthHandlerDelegateCallsBackWhenAuthenticatingDirectlyIfRefreshing]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.AlamofireClientTests testForwardsSendRequestToAlamofireAndRespectsHeaders]' started.
Test Case '-[AutoGraphTests.AlamofireClientTests testForwardsSendRequestToAlamofireAndRespectsHeaders]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.AlamofireClientTests testSetsRetrierAndAdapterOnSession]' started.
Test Case '-[AutoGraphTests.AlamofireClientTests testSetsRetrierAndAdapterOnSession]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.AlamofireClientTests testUpdatesRetrierAndAdapterWithNewAuthHandler]' started.
Test Case '-[AutoGraphTests.AlamofireClientTests testUpdatesRetrierAndAdapterWithNewAuthHandler]' passed (0.000 seconds).
Test Suite 'AlamofireClientTests' passed at 2018-07-18 14:04:59.216.
	 Executed 5 tests, with 0 failures (0 unexpected) in 0.145 (0.145) seconds
Test Suite 'AuthHandlerTests' started at 2018-07-18 14:04:59.216
Test Case '-[AutoGraphTests.AuthHandlerTests testAdaptsAuthToken]' started.
Test Case '-[AutoGraphTests.AuthHandlerTests testAdaptsAuthToken]' passed (0.004 seconds).
Test Case '-[AutoGraphTests.AuthHandlerTests testDoesNotGetAuthTokensIfFailure]' started.
Test Case '-[AutoGraphTests.AuthHandlerTests testDoesNotGetAuthTokensIfFailure]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.AuthHandlerTests testGetsAuthTokensIfSuccess]' started.
Test Case '-[AutoGraphTests.AuthHandlerTests testGetsAuthTokensIfSuccess]' passed (0.000 seconds).
Test Suite 'AuthHandlerTests' passed at 2018-07-18 14:04:59.221.
	 Executed 3 tests, with 0 failures (0 unexpected) in 0.004 (0.004) seconds
Test Suite 'AutoGraphTests' started at 2018-07-18 14:04:59.221
Test Case '-[AutoGraphTests.AutoGraphTests testAuthHandlerBeganReauthenticationPausesDispatcher]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testAuthHandlerBeganReauthenticationPausesDispatcher]' passed (0.009 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testAuthHandlerReauthenticatedSuccessfullyUnpausesDispatcher]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testAuthHandlerReauthenticatedSuccessfullyUnpausesDispatcher]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testAuthHandlerReauthenticatedUnsuccessfullyCancelsAll]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testAuthHandlerReauthenticatedUnsuccessfullyCancelsAll]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testCancelAllCancelsDispatcherAndClient]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testCancelAllCancelsDispatcherAndClient]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testFunctional401Request]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testFunctional401Request]' passed (0.517 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testFunctional401RequestNotHandled]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testFunctional401RequestNotHandled]' passed (0.517 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalAllFilmsRequest]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalAllFilmsRequest]' passed (0.516 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalGlobalLifeCycle]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalGlobalLifeCycle]' passed (0.513 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalLifeCycle]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalLifeCycle]' passed (0.517 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalSingleFilmRequest]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testFunctionalSingleFilmRequest]' passed (0.518 seconds).
Test Case '-[AutoGraphTests.AutoGraphTests testTriggeringReauthenticationPausesSystem]' started.
Test Case '-[AutoGraphTests.AutoGraphTests testTriggeringReauthenticationPausesSystem]' passed (0.015 seconds).
Test Suite 'AutoGraphTests' passed at 2018-07-18 14:05:02.345.
	 Executed 11 tests, with 0 failures (0 unexpected) in 3.124 (3.125) seconds
Test Suite 'DispatcherTests' started at 2018-07-18 14:05:02.346
Test Case '-[AutoGraphTests.DispatcherTests testClearsRequestsOnCancel]' started.
Test Case '-[AutoGraphTests.DispatcherTests testClearsRequestsOnCancel]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.DispatcherTests testFailureReturnsToCaller]' started.
Test Case '-[AutoGraphTests.DispatcherTests testFailureReturnsToCaller]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.DispatcherTests testForwardsAndClearsPendingRequestsOnUnpause]' started.
Test Case '-[AutoGraphTests.DispatcherTests testForwardsAndClearsPendingRequestsOnUnpause]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.DispatcherTests testForwardsRequestToSender]' started.
Test Case '-[AutoGraphTests.DispatcherTests testForwardsRequestToSender]' passed (0.001 seconds).
Test Case '-[AutoGraphTests.DispatcherTests testHoldsRequestsWhenPaused]' started.
Test Case '-[AutoGraphTests.DispatcherTests testHoldsRequestsWhenPaused]' passed (0.000 seconds).
Test Suite 'DispatcherTests' passed at 2018-07-18 14:05:02.348.
	 Executed 5 tests, with 0 failures (0 unexpected) in 0.003 (0.003) seconds
Test Suite 'DocumentTests' started at 2018-07-18 14:05:02.348
Test Case '-[QueryBuilderTests.DocumentTests testGraphQLStringWithObjectFieldsFragments]' started.
Test Case '-[QueryBuilderTests.DocumentTests testGraphQLStringWithObjectFieldsFragments]' passed (0.001 seconds).
Test Suite 'DocumentTests' passed at 2018-07-18 14:05:02.349.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'ErrorTests' started at 2018-07-18 14:05:02.349
Test Case '-[AutoGraphTests.ErrorTests testAutoGraphErrorGraphQLErrorUsesMessages]' started.
Test Case '-[AutoGraphTests.ErrorTests testAutoGraphErrorGraphQLErrorUsesMessages]' passed (0.002 seconds).
Test Case '-[AutoGraphTests.ErrorTests testAutoGraphErrorProducesNetworkErrorForNetworkErrorParserMatch]' started.
Test Case '-[AutoGraphTests.ErrorTests testAutoGraphErrorProducesNetworkErrorForNetworkErrorParserMatch]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.ErrorTests testGraphQLErrorUsesMessageForLocalizedDescription]' started.
Test Case '-[AutoGraphTests.ErrorTests testGraphQLErrorUsesMessageForLocalizedDescription]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.ErrorTests testInvalidResponseLocalizedErrorDoesntCrash]' started.
Test Case '-[AutoGraphTests.ErrorTests testInvalidResponseLocalizedErrorDoesntCrash]' passed (0.001 seconds).
Test Suite 'ErrorTests' passed at 2018-07-18 14:05:02.353.
	 Executed 4 tests, with 0 failures (0 unexpected) in 0.003 (0.003) seconds
Test Suite 'FieldTests' started at 2018-07-18 14:05:02.353
Test Case '-[QueryBuilderTests.FieldTests testSerializeAlias]' started.
Test Case '-[QueryBuilderTests.FieldTests testSerializeAlias]' passed (0.000 seconds).
Test Suite 'FieldTests' passed at 2018-07-18 14:05:02.353.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.000 (0.000) seconds
Test Suite 'FoundationExtensionsTests' started at 2018-07-18 14:05:02.353
Test Case '-[QueryBuilderTests.FoundationExtensionsTests testArgumentStringJsonEncodes]' started.
Test Case '-[QueryBuilderTests.FoundationExtensionsTests testArgumentStringJsonEncodes]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FoundationExtensionsTests testNSNullJsonEncodes]' started.
Test Case '-[QueryBuilderTests.FoundationExtensionsTests testNSNullJsonEncodes]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FoundationExtensionsTests testNSNumberJsonEncodes]' started.
Test Case '-[QueryBuilderTests.FoundationExtensionsTests testNSNumberJsonEncodes]' passed (0.000 seconds).
Test Suite 'FoundationExtensionsTests' passed at 2018-07-18 14:05:02.354.
	 Executed 3 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'FragmentDefinitionTests' started at 2018-07-18 14:05:02.354
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testFragmentNamedOnIsNil]' started.
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testFragmentNamedOnIsNil]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithDirectives]' started.
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithDirectives]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithFragments]' started.
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithFragments]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithObjectFields]' started.
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithObjectFields]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithScalarFields]' started.
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testGraphQLStringWithScalarFields]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testSelectionSetName]' started.
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testSelectionSetName]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testWithoutSelectionSetIsNil]' started.
Test Case '-[QueryBuilderTests.FragmentDefinitionTests testWithoutSelectionSetIsNil]' passed (0.000 seconds).
Test Suite 'FragmentDefinitionTests' passed at 2018-07-18 14:05:02.356.
	 Executed 7 tests, with 0 failures (0 unexpected) in 0.001 (0.002) seconds
Test Suite 'InlineFragmentTests' started at 2018-07-18 14:05:02.356
Test Case '-[QueryBuilderTests.InlineFragmentTests testInlineFragment]' started.
Test Case '-[QueryBuilderTests.InlineFragmentTests testInlineFragment]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InlineFragmentTests testObjectWithInlineFragment]' started.
Test Case '-[QueryBuilderTests.InlineFragmentTests testObjectWithInlineFragment]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InlineFragmentTests testSelectionSetName]' started.
Test Case '-[QueryBuilderTests.InlineFragmentTests testSelectionSetName]' passed (0.000 seconds).
Test Suite 'InlineFragmentTests' passed at 2018-07-18 14:05:02.357.
	 Executed 3 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'InputValueTests' started at 2018-07-18 14:05:02.357
Test Case '-[QueryBuilderTests.InputValueTests testArrayInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testArrayInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testBoolInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testBoolInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testDictionaryInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testDictionaryInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testDoubleInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testDoubleInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testEmptyArrayInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testEmptyArrayInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testEmptyDictionaryInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testEmptyDictionaryInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testEnumInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testEnumInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testIDInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testIDInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testIntInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testIntInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testNonNullInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testNonNullInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testNSNullInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testNSNullInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.InputValueTests testVariableInputValue]' started.
Test Case '-[QueryBuilderTests.InputValueTests testVariableInputValue]' passed (0.000 seconds).
Test Suite 'InputValueTests' passed at 2018-07-18 14:05:02.360.
	 Executed 12 tests, with 0 failures (0 unexpected) in 0.002 (0.003) seconds
Test Suite 'ObjectTests' started at 2018-07-18 14:05:02.360
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithAlias]' started.
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithAlias]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithObjectFields]' started.
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithObjectFields]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithoutAlias]' started.
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithoutAlias]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithScalarFields]' started.
Test Case '-[QueryBuilderTests.ObjectTests testGraphQLStringWithScalarFields]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.ObjectTests testThrowsIfNoFieldsOrFragments]' started.
Test Case '-[QueryBuilderTests.ObjectTests testThrowsIfNoFieldsOrFragments]' passed (0.000 seconds).
Test Suite 'ObjectTests' passed at 2018-07-18 14:05:02.361.
	 Executed 5 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'OperationTests' started at 2018-07-18 14:05:02.361
Test Case '-[QueryBuilderTests.OperationTests testDirectives]' started.
Test Case '-[QueryBuilderTests.OperationTests testDirectives]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.OperationTests testInitializersOnSelectionTypeArray]' started.
Test Case '-[QueryBuilderTests.OperationTests testInitializersOnSelectionTypeArray]' passed (0.001 seconds).
Test Case '-[QueryBuilderTests.OperationTests testMutationForms]' started.
Test Case '-[QueryBuilderTests.OperationTests testMutationForms]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.OperationTests testQueryForms]' started.
Test Case '-[QueryBuilderTests.OperationTests testQueryForms]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.OperationTests testSelectionSet]' started.
Test Case '-[QueryBuilderTests.OperationTests testSelectionSet]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.OperationTests testSubscriptionForms]' started.
Test Case '-[QueryBuilderTests.OperationTests testSubscriptionForms]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.OperationTests testVariableDefinitions]' started.
Test Case '-[QueryBuilderTests.OperationTests testVariableDefinitions]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.OperationTests testVariableVariablesWithDefaultValuesFail]' started.
Test Case '-[QueryBuilderTests.OperationTests testVariableVariablesWithDefaultValuesFail]' passed (0.000 seconds).
Test Suite 'OperationTests' passed at 2018-07-18 14:05:02.364.
	 Executed 8 tests, with 0 failures (0 unexpected) in 0.003 (0.003) seconds
Test Suite 'OrderedDictionaryTests' started at 2018-07-18 14:05:02.364
Test Case '-[QueryBuilderTests.OrderedDictionaryTests testsMaintainsOrder]' started.
Test Case '-[QueryBuilderTests.OrderedDictionaryTests testsMaintainsOrder]' passed (0.001 seconds).
Test Suite 'OrderedDictionaryTests' passed at 2018-07-18 14:05:02.365.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'ResponseHandlerTests' started at 2018-07-18 14:05:02.365
Test Case '-[AutoGraphTests.ResponseHandlerTests testErrorsJsonReturnsGraphQLError]' started.
Test Case '-[AutoGraphTests.ResponseHandlerTests testErrorsJsonReturnsGraphQLError]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.ResponseHandlerTests testMappingErrorReturnsMappingError]' started.
Test Case '-[AutoGraphTests.ResponseHandlerTests testMappingErrorReturnsMappingError]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.ResponseHandlerTests testNetworkErrorReturnsNetworkError]' started.
Test Case '-[AutoGraphTests.ResponseHandlerTests testNetworkErrorReturnsNetworkError]' passed (0.000 seconds).
Test Case '-[AutoGraphTests.ResponseHandlerTests testPreMappingHookCalledBeforeMapping]' started.
Test Case '-[AutoGraphTests.ResponseHandlerTests testPreMappingHookCalledBeforeMapping]' passed (0.000 seconds).
Test Suite 'ResponseHandlerTests' passed at 2018-07-18 14:05:02.367.
	 Executed 4 tests, with 0 failures (0 unexpected) in 0.001 (0.001) seconds
Test Suite 'ScalarTests' started at 2018-07-18 14:05:02.367
Test Case '-[QueryBuilderTests.ScalarTests testGraphQLStringAsLiteral]' started.
Test Case '-[QueryBuilderTests.ScalarTests testGraphQLStringAsLiteral]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.ScalarTests testGraphQLStringWithAlias]' started.
Test Case '-[QueryBuilderTests.ScalarTests testGraphQLStringWithAlias]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.ScalarTests testGraphQLStringWithoutAlias]' started.
Test Case '-[QueryBuilderTests.ScalarTests testGraphQLStringWithoutAlias]' passed (0.000 seconds).
Test Suite 'ScalarTests' passed at 2018-07-18 14:05:02.367.
	 Executed 3 tests, with 0 failures (0 unexpected) in 0.000 (0.001) seconds
Test Suite 'SelectionSetTests' started at 2018-07-18 14:05:02.367
Test Case '-[QueryBuilderTests.SelectionSetTests testGraphQLString]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testGraphQLString]' passed (0.001 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testKey]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testKey]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testKind]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testKind]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingFragmentSpreads]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingFragmentSpreads]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingScalars]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingScalars]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingSelections]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingSelections]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingSelectionsOfSameKeyButDifferentTypeFails]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testMergingSelectionsOfSameKeyButDifferentTypeFails]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testSelectionSetName]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testSelectionSetName]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.SelectionSetTests testSerializedSelections]' started.
Test Case '-[QueryBuilderTests.SelectionSetTests testSerializedSelections]' passed (0.000 seconds).
Test Suite 'SelectionSetTests' passed at 2018-07-18 14:05:02.370.
	 Executed 9 tests, with 0 failures (0 unexpected) in 0.002 (0.003) seconds
Test Suite 'VariableTest' started at 2018-07-18 14:05:02.370
Test Case '-[QueryBuilderTests.VariableTest testVariableInputValue]' started.
Test Case '-[QueryBuilderTests.VariableTest testVariableInputValue]' passed (0.000 seconds).
Test Case '-[QueryBuilderTests.VariableTest testVariableTypeThrows]' started.
Test Case '-[QueryBuilderTests.VariableTest testVariableTypeThrows]' passed (0.000 seconds).
Test Suite 'VariableTest' passed at 2018-07-18 14:05:02.370.
	 Executed 2 tests, with 0 failures (0 unexpected) in 0.000 (0.000) seconds
Test Suite 'AutoGraphPackageTests.xctest' passed at 2018-07-18 14:05:02.370.
	 Executed 87 tests, with 0 failures (0 unexpected) in 3.294 (3.299) seconds
Test Suite 'All tests' passed at 2018-07-18 14:05:02.371.
	 Executed 87 tests, with 0 failures (0 unexpected) in 3.294 (3.300) seconds
```

### CODE REVIEW

File: AutoGraphTests/Stub.swift [hunk 1]
```diff
@@ -1,10 +1,36 @@
 import Foundation
-import OHHTTPStubs
+
```
Comment (rex-remind101) [inline AutoGraphTests/Stub.swift]: OHHTTPStubs requires Obj-C, so I had to write a new network stubbing system that was pure Swift which is most of the changes in this file.

File: AutoGraphTests/Stub.swift [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+            let path: String = {
+                #if os(iOS)
+                return Bundle(for: type(of: self)).path(forResource: jsonFixtureFile, ofType: "json")!
+                
+                #else
+                let fileManager = FileManager.default
+                let currentDirectoryPath = fileManager.currentDirectoryPath
+                return "\(currentDirectoryPath)/AutoGraphTests/Data/\(jsonFixtureFile).json"
+                
+                #endif
```
Comment (rex-remind101) [inline AutoGraphTests/Stub.swift]: iOS has a Bundle, but command line swift does not. This gets the right path for the stubbed data.

File: AutoGraphTests/Stub.swift [hunk 3]
```diff
@@ -112,3 +129,102 @@ extension String {
...
+            if amount > 0 {
+                result.append(buffer, count: amount)
+            }
+        } while amount > 0
+        
+        return result
+    }
+}
+
+class MockURLProtocol: URLProtocol {
```
Comment (rex-remind101) [inline AutoGraphTests/Stub.swift]: `MockURLProtocol` hijacks the URLProtocol that a request will use when a stub is registered. Everything ends up going through `startLoading`.

File: AutoGraphTests/Stub.swift [hunk 4]
```diff
@@ -112,3 +129,102 @@ extension String {
         return components.filter { !$0.isEmpty }.joined(separator: " ")
     }
 }
+
+extension InputStream {
```
Comment (rex-remind101) [inline AutoGraphTests/Stub.swift]: This is needed for https://github.com/remind101/AutoGraph/pull/84/files#diff-9ed08f48421332f9614168f59bb22ce6R84

Unfortunately there's a bug with URLProtocol where by the time the request is handed off to you the data is already in `httpBodyStream` and `httpBody` is set to `nil`. So we use this to read the body back out into `Data`.

File: AutoGraphTests/XCTestManifests.swift [hunk 5]
```diff
@@ -0,0 +1,76 @@
+import XCTest
+
+extension AlamofireClientTests {
```
Comment (rex-remind101) [inline AutoGraphTests/XCTestManifests.swift]: This is all generated boilerplate for Linux tests.

File: Package.swift [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// swift-tools-version:4.0
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
+
+let package = Package(
```
Comment (rex-remind101) [inline Package.swift]: This is our Package file for SPM

File: README.md [hunk 7]
```diff
@@ -51,8 +51,12 @@ use_frameworks!
...
 pod 'AutoGraph'
 ```
-### Future
-- [ ] Swift Package Manager
+### Swift Package Manager (SPM)
+```swift
+dependencies: [
+.package(url: "https://github.com/remind101/AutoGraph.git", .upToNextMinor(from: "0.5.2"))
+]
+```
```
Comment (rex-remind101) [inline README.md]: README stuff

File: AutoGraphTests/XCTestManifests.swift [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    static let __allTests = [
+        ("testAdaptsAuthToken", testAdaptsAuthToken),
+        ("testDoesNotGetAuthTokensIfFailure", testDoesNotGetAuthTokensIfFailure),
+        ("testGetsAuthTokensIfSuccess", testGetsAuthTokensIfSuccess),
+    ]
+}
+
+extension AutoGraphTests {
+    static let __allTests = [
+        ("testAuthHandlerBeganReauthenticationPausesDispatcher", testAuthHandlerBeganReauthenticationPausesDispatcher),
```
Comment (jessieserrino) [inline AutoGraphTests/XCTestManifests.swift]: Seems that these tests are failing

Comment (jessieserrino) [reply AutoGraphTests/XCTestManifests.swift]: Once you fix that, it's good though

Comment (rex-remind101) [reply AutoGraphTests/XCTestManifests.swift]: clearing the ci cache fixed it

**Pull request merged**